src/pretix/plugins/sendmail/signals.py

```diff
@@ -65,11 +65,10 @@ def scheduled_mail_create(sender, **kwargs):
     with scope(organizer=event.organizer):
         existing_rules = ScheduledMail.objects.filter(subevent=subevent).values_list('rule_id', flat=True)
         to_create = []
-        for rule in event.sendmail_rules.all():
-            if rule.pk not in existing_rules and subevent:
-                sm = ScheduledMail(rule=rule, event=event, subevent=subevent)
-                sm.recompute()
-                to_create.append(sm)
+        for rule in event.sendmail_rules.filter(subevent=None).exclude(id__in=existing_rules):
```

Comment:
what about the subevent=subevent case?

Reply:
The subevent=subevent-case would not need a new `ScheduledMail`. If it is a newly created subevent, there cannot be any sendmail_rule containing it, so we only need to add ScheduledMails to rules that apply to all (and all future) subevents (therefore subevent=None). If the subevent already existed and is updated and sendmail_rules exist for it, then there also should already exist ScheduledMails for these sendmail_rules, as they got created when the sendmail_rule was created.

Recomputation of existing ScheduledMails due to changes in subevents regarding date and time is done in the periodic task `sendmail_run_rules`. We could of course do it here as well, but it wasn’t before, so I thought I leave it that way.

Reply:
Okay, makes sense